### PR TITLE
Terraform 0.12 support

### PR DESCRIPTION
**Pull request by osterman**
## what
* Port (rewrite) module with 0.12 syntax

## why
* Module currently does not work with 0.12
* Much easier syntax

## todo
- [x] update tests
- [x] update CI to support 0.12
- [x] verify identical outputs
- [x] update README 
- [x] update README examples

### CODE REVIEW

File: examples/autoscalinggroup/versions.tf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+terraform {
+  required_version = ">= 0.12"
```
Comment (Nuru) [inline examples/autoscalinggroup/versions.tf]:
```suggestion
  required_version = "~> 0.12.0"
```

File: examples/complete/versions.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+terraform {
+  required_version = ">= 0.12"
```
Comment (Nuru) [inline examples/complete/versions.tf]:
```suggestion
  required_version = "~> 0.12.0"
```

File: main.tf [hunk 3]
```diff
@@ -1,98 +1,72 @@
...
+    delimiter   = "-"
+    replacement = ""
+    # The `sentinel` should match the `regex_replace_chars`, so it will be replaced with the `replacement` value
+    sentinel   = "~"
+    attributes = [""]
   }
 
-  # Merge the map of empty values, with the variable context, so that context_local always contains all map keys
-  context_local = "${merge(local.context_struct, var.context)}"
+  # Provided values provided by variables superceed values inherited from the context
```
Comment (aknysh) [inline main.tf]:
```suggestion
  # The values provided by variables superceed the values inherited from the context
```

File: main.tf [hunk 4]
```diff
@@ -1,98 +1,72 @@
...
-    )}"
-  tags                     = "${merge(zipmap(local.context_local["tags_keys"], local.context_local["tags_values"]), local.generated_tags, var.tags)}"
-  tags_as_list_of_maps     = ["${data.null_data_source.tags_as_list_of_maps.*.outputs}"]
-  label_order_default_list = "${list("namespace", "environment", "stage", "name", "attributes")}"
-  label_order_context_list = "${distinct(compact(local.context_local["label_order"]))}"
-  label_order_final_list   = ["${distinct(compact(coalescelist(var.label_order, local.label_order_context_list, local.label_order_default_list)))}"]
-  label_order_length       = "${(length(local.label_order_final_list))}"
+  attributes = compact(distinct(concat(var.attributes, var.context.attributes, local.defaults.attributes)))
+
+  generated_tags = { for l in keys(local.id_context) : title(l) => local.id_context[l] if length(local.id_context[l]) > 0 }
```
Comment (aknysh) [inline main.tf]: wow, TF 0.12 shines here :)

File: test/src/Makefile [hunk 5]
```diff
@@ -1,42 +1,50 @@
...
+## Install go, if not installed
+$(GOEXE):
+	apk add --update go
+endif
 
-export GOPATH
+ifeq ($(shell uname -s),Linux)
+## Install all `dep`, if not installed
+$(GOBIN)/dep:
+	mkdir -p $(GOBIN)
```
Comment (aknysh) [inline test/src/Makefile]:
```suggestion
	@mkdir -p $(GOBIN)
```

as in other places

File: main.tf (hunk 4 above)
Comment (osterman) [reply main.tf]: Yes, I liked this. Reminded me of the `map { ... }` expression in ruby and perl.

**Pull request merged**